apps/java-api/src/main/java/org/scoalaonline/api/service/GradeService.java

```diff
@@ -0,0 +1,94 @@
+package org.scoalaonline.api.service;
+
+import org.scoalaonline.api.exception.grade.GradeInvalidValueException;
+import org.scoalaonline.api.exception.grade.GradeNotFoundException;
+import org.scoalaonline.api.model.Grade;
+import org.scoalaonline.api.repository.GradeRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+
+/**
+ * Contains the Grade related logic needed for the API
+ */
+@Service
+public class GradeService implements ServiceInterface<Grade> {
+  @Autowired
+  GradeRepository gradeRepository;
+
+  /**
+   * Retrieves a list of all Grade entries found in the DB
+   * @return the list of Grade entries
+   */
+  @Override
+  public List<Grade> getAll() {
+    return gradeRepository.findAll();
+  }
+
+  /**
+   * Retrieves one Grade entry with the given id from the DB
+   * or throws an error if no entry with that id is found.
+   * @param id - id of the Grade entry
+   * @return the Grade entry
+   * @throws GradeNotFoundException
+   */
+  @Override
+  public Grade getOneById(String id) throws GradeNotFoundException {
+    return gradeRepository.findById(id).orElseThrow(
+      () -> new GradeNotFoundException("Method getOneById: Grade not found")
+    );
+  }
+
+  /**
+   * Adds a Grade entry in the DB based on the received object.
+   * @param entry
+   * @throws GradeInvalidValueException
+   * @return the Grade object that has been saved in the DB
+   */
+  @Override
+  public Grade add(Grade entry) throws GradeInvalidValueException {
+    Grade gradeToSave = new Grade();
+    if(entry.getValue() > 0 ) {
```

Comment:
Perhaps also add an upper limit of `12`, so we would only have values from `1` to `12`.

Reply:
Don't forget to apply it to `update` method too.